test/page-autowaiting-basic.spec.ts

```diff
@@ -120,63 +160,30 @@ it('should await navigation when assigning location twice', async ({page, server
 });
 
 it('should await navigation when evaluating reload', async ({page, server}) => {
-  const messages = [];
   await page.goto(server.EMPTY_PAGE);
-  server.setRoute('/empty.html', async (req, res) => {
-    messages.push('route');
-    res.setHeader('Content-Type', 'text/html');
-    res.end(`<link rel='stylesheet' href='./one-style.css'>`);
-  });
-
+  const messages = initServer(server);
   await Promise.all([
     page.evaluate(`window.location.reload()`).then(() => messages.push('evaluate')),
     page.waitForEvent('framenavigated').then(() => messages.push('navigated')),
   ]);
   expect(messages.join('|')).toBe('route|navigated|evaluate');
 });
 
-it('should await navigating specified target', async ({page, server}) => {
-  const messages = [];
-  server.setRoute('/empty.html', async (req, res) => {
-    messages.push('route');
-    res.setHeader('Content-Type', 'text/html');
-    res.end(`<link rel='stylesheet' href='./one-style.css'>`);
-  });
-
-  await page.setContent(`
-    <a href="${server.EMPTY_PAGE}" target=target>empty.html</a>
-    <iframe name=target></iframe>
-  `);
-  const frame = page.frame({ name: 'target' });
-  await Promise.all([
-    page.click('a').then(() => messages.push('click')),
-    page.waitForEvent('framenavigated').then(() => messages.push('navigated')),
```

Comment:
can we have a test that checks that we don't wait for the frame navigation ?

Reply:
It might be tricky - we would need to simulate slow iframe navigation commit :/